code/modules/clothing/shoes/magboots.dm

```diff
@@ -53,4 +53,25 @@
 obj/item/clothing/shoes/magboots/syndie/advance //For the Syndicate Strike Team
 	desc = "Reverse-engineered magboots that appear to be based on an advanced model, as they have a lighter magnetic pull. Property of Gorlex Marauders."
 	name = "advanced blood-red magboots"
-	slowdown_active = SHOES_SLOWDOWN
\ No newline at end of file
+	slowdown_active = SHOES_SLOWDOWN
+
+/obj/item/clothing/shoes/magboots/clown
+	desc = "The prankster's standard-issue clowning shoes. Damn they're huge! There's a red light on the side."
+	name = "clown shoes"
+	icon_state = "clown"
+	item_state = "clown_shoes"
+	slowdown = SHOES_SLOWDOWN+1
+	item_color = "clown"
+	var/footstep = 1	//used for squeeks whilst walking
+	silence_steps = 1
+	shoe_sound = "clownstep"
+	origin_tech = "magnets=4;syndicate=2"
+
+/obj/item/clothing/shoes/magboots/clown/attack_self(mob/user) //redefine toggle to remove slowdown and icon updating

```

Comment:
Is this necessary, given that they are a child object of the shoes/magboots?
Also, what happens if you toggle their status? Does it change their icon, like normal magboots do? Is it meant to?

Reply:
The issue is that the original magboots toggle reset the player's speed back to default instead of back to whatever the previous speed was. I could fix this by adding a slowdown_passive var and modify the toggle accordingly, but this would be modifying the code of the original magboots, which I wasn't 100% comfortable with given this is my first PR. This felt like a better solution, although it does duplicate a little code.

When you toggle the shoes, the icon is not changed. This is intended behaviour, so that the shoes are not easily spotted if removed unless they are actively inspected. The player does get the message "You activate/disable the mag-pulse traction system", so although they'll have to be more careful, they know whether it's disabled or enabled. You can also examine the shoes while equipped, and it'll tell you the status of the mag-pulse.

Reply:
I've not played with magboot code myself but couldn't you just do `slowdown_active = SHOES_SLOWDOWN` and not have to update the `attack_self` proc? (Similar to how advance magboots does it in https://github.com/datlo/Paradise/blob/8a909e4da5a5a443dd45af4ec9f64592e4537108/code/modules/clothing/shoes/magboots.dm#L38)

Or even just make these clown magboots a subtype of `obj/item/clothing/shoes/magboots/advance`

Reply:
Unless I'm misunderstanding you, this doesn't work, because the original magboots proc returns the speed to default speed when deactivated. We want the proc to return to SHOES_SLOWDOWN+1, or, as I did it, cut out slowing up and down on activation entirely.

Reply:
To reduce code duplication, could you maybe call the parent proc and then return it to the correct slowdown after?

Reply:
In that case, do something like the following:

```
/obj/item/clothing/shoes/magboots/clown/attack_self(mob/user)
    ..()
    if(!magpulse)
        slowdown = initial(slowdown)
```
This prevents duplicating code whilst also resetting the slowdown